src/kratos.ts

```diff
@@ -59,75 +60,78 @@ export const clearAuthentication = async (
-  setTrackingId(req, res, req.trackingId);
-  setCookie(req, res, 'auth', undefined);
-  setCookie(req, res, 'kratosContinuity', undefined);
-  setCookie(req, res, 'kratos', undefined);
-};
+): Promise<void> =>
+  runInSpan('clearAuthentication', async () => {
+    req.log.info(
+      {
+        reason,
+        userId: req.userId,
+        cookie: req.cookies[cookies.kratos.key],
+      },
+      'clearing authentication',
+    );
+    req.trackingId = await generateTrackingId();
+    req.userId = undefined;
+    setTrackingId(req, res, req.trackingId);
+    setCookie(req, res, 'auth', undefined);
+    setCookie(req, res, 'kratosContinuity', undefined);
+    setCookie(req, res, 'kratos', undefined);
+  });
 
 type WhoamiResponse =
   | { valid: true; userId: string; expires: Date; cookie?: string }
   | { valid: false };
 
 export const dispatchWhoami = async (
   req: FastifyRequest,
-): Promise<WhoamiResponse> => {
-  if (heimdallOrigin === 'disabled' || !req.cookies[cookies.kratos.key]) {
-    return { valid: false };
-  }
-  try {
-    const { res: whoami, headers } = await fetchKratos(
-      req,
-      `${heimdallOrigin}/api/whoami`,
-    );
-    if (whoami?.identity?.traits?.userId) {
-      return {
-        valid: true,
-        userId: whoami.identity.traits.userId,
-        expires: new Date(whoami.expires_at),
-        cookie: headers.get('set-cookie'),
-      };
+): Promise<WhoamiResponse> =>
+  runInSpan('dispatchWhoami', async () => {
+    if (heimdallOrigin === 'disabled' || !req.cookies[cookies.kratos.key]) {
+      return { valid: false };
     }
-    req.log.info({ whoami }, 'invalid whoami response');
-  } catch (e) {
-    if (e.statusCode !== 401) {
-      throw e;
+    try {
+      const { res: whoami, headers } = await fetchKratos(
+        req,
+        `${heimdallOrigin}/api/whoami`,
+      );
+      if (whoami?.identity?.traits?.userId) {
+        return {
+          valid: true,
+          userId: whoami.identity.traits.userId,
+          expires: new Date(whoami.expires_at),
+          cookie: headers.get('set-cookie'),
+        };
+      }
+      req.log.info({ whoami }, 'invalid whoami response');
+    } catch (e) {
+      if (e.statusCode !== 401) {
+        throw e;
+      }
     }
-  }
 
-  return { valid: false };
-};
+    return { valid: false };
+  });
 
 export const logout = async (
   req: FastifyRequest,
   res: FastifyReply,
-): Promise<FastifyReply> => {
-  try {
-    const { res: logoutFlow } = await fetchKratos(
-      req,
-      `${kratosOrigin}/self-service/logout/browser`,
-    );
-    if (logoutFlow?.logout_url) {
-      const logoutParts = logoutFlow.logout_url.split('/self-service/');
-      const logoutUrl = `${kratosOrigin}/self-service/${logoutParts[1]}`;
-      await fetchKratos(req, logoutUrl, { redirect: 'manual' });
-    }
-  } catch (e) {
-    if (e.statusCode !== 303 && e.statusCode !== 401) {
-      throw e;
+): Promise<FastifyReply> =>
+  runInSpan('logout', async () => {
```

Comment:
Since we have one for `retryFetch` this feels redundant

Reply:
Yeah, I'll remove it